libcloud/compute/drivers/verizon.py

```diff
@@ -0,0 +1,890 @@
+        :type    name: ``str``
+
+        :keyword size: the plan size to create (mandatory)
+        :type    size: :class:`NodeSize`
+
+        :keyword image: Which image to use
+        :type    image: :class:`NodeImage` (mandatory)
+
+        :keyword ssh_admin_key: SSH key to inject for the admin user. 
+        :type    ssh_admin_key: ``str``
+
+        :keyword pool: Compute pool to deploy in
+        :type    pool: ``str``
+
+        :keyword network: Network to attach
+        :type       pool: ``str``
+
+        :keyword ipaddr: IP address to use
+        :type    ipaddr: ``str``
+
+        :keyword row: Row to deploy on
+        :type    row: ``str``
+
+        :keyword group: Group to deploy on
+        :type    group: ``str``
+
+        :return: Node object of newly created node
+        :rtype: :class: `Node`
+
+        """
+        if name is None:
+            raise VerizonException('Parameter name is required')
+        if image is None:
+            raise VerizonException('Parameter image is required')
+        if size is None:
+            raise VerizonException('Parameter size is required')
+
+        layout_hrefs = {}
+
+        if row is None and group is None:
+            layout_hrefs = self._libcloud_layout_check()
+        else:
+            layout_hrefs['row'] = row
+            layout_hrefs['group'] = group
+
+        if ssh_admin_key is None:
+            ssh_admin_key = self.ex_get_ssh_key()
+
+        if pool is None:
+            pool = self.ex_get_default_pool()
+
+        if network is None:
+            network = self.ex_get_network(type='int')
+            ipaddr = self.ex_get_address(network_href=network)
+            if ipaddr is None:
+                raise VerizonException('No ips are available')
+
+        vm = {}
+        vm['name'] = name
+        vm['PoweredOn'] = False
+        vm['ProcessorCount'] = size.cpu
+        vm['Template'] = {'href': image.id}
+
+        vm['Memory'] = {'Unit': 'MB',
+                        'Value': size.ram}
+
+        vm['Layout'] = {'Row': {'href': layout_hrefs['row']},
+                        'Group': {'href': layout_hrefs['group']}}
+
+        vm['LinuxCustomization'] = {'SshKey': {'href': ssh_admin_key},
+                                    'NetworkSettings': {
+                                        'NetworkAdapterSettings': [{
+                                            'Network': {'href': network},
+                                            'IpAddress': ipaddr}]
+                                        }
+                                    }
+
+        url = '%s/virtualmachines/computepools/%s/action/createvirtualmachine' % (API_ROOT, pool)
+
+        headers = {'Content-Length': len(json.dumps(vm))}
+
+        result = self.connection.request(url, method='POST', headers=headers, data=json.dumps(vm)).object
+        result['ip'] = ipaddr
+        return self._to_node(result)
+
+    def deploy_node(self, deploy=None, **kwargs):
```

Comment:
How is this different from base `deploy_node` method?

I'm not a fan of duplicating that much functionality, especially since this method already caused a lot of issues in the past...